hy/compiler.py

```diff
@@ -2163,7 +2212,12 @@ def nan(): return asty.BinOp(
 
     @builds(HySymbol)
     def compile_symbol(self, symbol):
-        if "." in symbol:
+        if symbol == "...":
+            if not PY3:
+                raise HyTypeError(symbol, 'ellipsis is only supported in '
+                                          'slices in Python 2')
```

Comment:
The error message might be better worded as "in Python 2, ellipsis is only supported in slices".